platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -1027,9 +1028,25 @@ private boolean hasValidCookie(HttpServletRequest request, boolean forceAuthn) {
         SecurityToken securityToken = new SecurityToken(assertionId, samlToken, null);
         SecurityAssertionImpl assertion = new SecurityAssertionImpl(securityToken);
 
-        if (forceAuthn || !assertion.isPresentlyValid()) {
+        if (forceAuthn) {
           cookieCache.removeSamlAssertion(key);
           return false;
+        } else if (!assertion.isPresentlyValid()) {
+          SAMLAuthenticationToken samlAuthenticationToken =
+              new SAMLAuthenticationToken(null, securityToken, "idp");
```

Comment:
✏️ It may be out of scope, but should something important like the name of the idp realm (`"idp"`) be refactored out into a constant?

Reply:
The realm thing is probably something we should rethink entirely for DDF. Seems like it causes more confusion than anything else at this point.